### DISCUSSION TITLE
Variables

### DISCUSSION THREAD

**Discussion by SoltauFintel** [facebook/docusaurus#9673] [General] [answered]
How can I add variables (e.g. version number) to my markdown files? I can't find anything about this in the Docusaurus documentation.

```
  "dependencies": {
    "@docusaurus/core": "3.0.1",
    "@docusaurus/preset-classic": "3.0.1",
```

**Comment by slorber** [comment, marked as answer]
It is unclear what it means to "add variables", please tell us where you want to add variables.

You might be interested to read:
- variables in markdown: https://github.com/facebook/docusaurus/issues/395
- variables in front matter: https://github.com/facebook/docusaurus/pull/9624
- variables accessible in Docusaurus context: https://docusaurus.io/docs/next/api/docusaurus-config#customFields

**Reply by SoltauFintel** [reply]
I did it [this way](https://github.com/facebook/docusaurus/issues/395#issuecomment-925306573). I added import command in markdown file:

```
---
sidebar_position: 1
---
import siteConfig from '/docusaurus.config.js';

# Getting started guide
... {siteConfig.customFields.x_version} ...
```

customFields in docusaurus.config.js:

```
...
const config = {
  title: '...',
  customFields: {
    x_version: '3.3',
  },
...
```

I can't get it work inside ´´´ code blocks ´´´.

**Reply by slorber** [reply]
@SoltauFintel you will need to process the code blocks with a remark plugin, as suggested in https://github.com/facebook/docusaurus/issues/395#issuecomment-1727777067
